sfge/src/main/java/com/salesforce/graph/build/AbstractApexVertexBuilder.java

```diff
@@ -66,11 +66,32 @@ private void buildVertices(JorjeNode node, Vertex vNodeParam, String fileName) {
             vNode.property(Schema.FILE_NAME, fileName);
         }
 
-        Vertex vPreviousSibling = null;
         final List<JorjeNode> children = node.getChildren();
+        // Synthetic vertices create the possibility that children will be processed out of order.
+        // So instead of using an ArrayList, we use a HashMap whose keys are the indices.
```

Comment:
i dont' understand.  do you intend to prevent out of order access of an element in from a collection ? in other words, do you intend to preserve the order of elements?

Reply:
Yeah, the goal is to preserve ordering.
The new kinds of synthetic vertex are created before the real children are processed, but some of them actually belong _after_ the real children in terms of strict ordering. ArrayLists don't let you have null indices, so the map basically lets us create the vertices out of order and sort it out later.

Reply:
that is why i got spooked.  your intention was to preserve order and yet changed the underlying data structure from List to Hash.  So long as you are aware and re-ordering as needed, that sound good